scripts/src/primazactl/primaza/primazacluster.py

```diff
@@ -51,44 +62,41 @@ def get_updated_server_url(self):
 
     def get_kubeconfig(self, identity: KubeIdentity,
                        other_cluster_name) -> Dict:
-        logger.log_entry(f"id: {identity.identity}, "
+        logger.log_entry(f"id: {identity.sa_name}, "
                          f"other_cluster_name: {other_cluster_name}")
         server_url = self.get_updated_server_url() \
             if self.cluster_name != other_cluster_name \
             else None
 
         return identity.get_kubeconfig(self.kubeconfig, server_url)
 
-    def create_identity(self, user: str = None) -> KubeIdentity:
+    def create_identity(self, sa_name: str, key_name: str) -> KubeIdentity:
         logger.log_entry()
-        if user:
-            self.user = user
         api_client = self.kubeconfig.get_api_client()
-        identity = KubeIdentity(api_client, self.user, self.namespace)
+        identity = KubeIdentity(api_client, sa_name, key_name, self.namespace)
         identity.create()
         return identity
 
-    def create_namespaced_secret(self, secret_name: str, kubeconfig: str):
+    def create_namespaced_secret(self, kubeconfig: str,
```

Comment:
to avoid confusion, I'd propose to rename this method such that it is clear that it will create a kubeconfig secret

```suggestion
    def create_namespaced_kubeconfig_secret(self, kubeconfig: str,
```

Reply:
Done.